### PR TITLE
[SPARK-49703] Publish Java 21 Docker image for preview1

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to publish Java 21 Docker image for `preview1` and will be extended for `preview2`.

### Why are the changes needed?

Apache Spark supports Java 21 via SPARK-43831.

### Does this PR introduce _any_ user-facing change?

No, this is a new image.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
cc @Yikun

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon .

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ You can see [3.4.0 PR](https://github.com/apache/spark-docker/pull/33) as refere
 
 ### Step 2. Publish apache/spark Images.
 
-Click [Publish (Java 17 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml) (such as 4.x) or [Publish](https://github.com/apache/spark-docker/actions/workflows/publish.yml) (such as 3.x) to publish images.
+Click [Publish (Java 21 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml), [Publish (Java 17 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml) (such as 4.x) or [Publish](https://github.com/apache/spark-docker/actions/workflows/publish.yml) (such as 3.x) to publish images.
```
Comment (viirya) [inline README.md:38]:
```suggestion
Click [Publish (Java 21 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java21.yaml), [Publish (Java 17 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml) (such as 4.x) or [Publish](https://github.com/apache/spark-docker/actions/workflows/publish.yml) (such as 3.x) to publish images.
```
```suggestion
Click [Publish (Java 21 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml), [Publish (Java 17 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml) (such as 4.x) or [Publish](https://github.com/apache/spark-docker/actions/workflows/publish.yml) (such as 3.x) to publish images.
```

**Comment by dongjoon-hyun** [comment]
For the record, while preparing `4.0.0-preview2` release, I'm making this repo up-to-date by re-implementing the Apache Spark patches like the following. The following three are published to https://hub.docker.com/r/apache/spark/ .
- #66 
- #67 
- #68

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya .

File: versions.json [hunk 2]
```diff
@@ -1,5 +1,34 @@
 {
   "versions": [
+    {
+      "path": "4.0.0-preview1/scala2.13-java21-python3-ubuntu",
+      "tags": [
+        "4.0.0-preview1-scala2.13-java21-python3-ubuntu",
+        "4.0.0-preview1-java21-python3",
+        "4.0.0-java21",
```
Comment (Yikun) [inline versions.json:8]: 4.0.0-preview1-java21

File: versions.json [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+        "python3-java21"
```
Comment (Yikun) [inline versions.json:9]: I'm okay with this, anther choice maybe better to add this after 4.0.0 published?

File: versions.json (hunk 2 above)
Comment (dongjoon-hyun) [reply versions.json:8]: Oh, right. Thank you!

File: versions.json (hunk 3 above)
Comment (dongjoon-hyun) [reply versions.json:9]: Let me remove this. :)

**Comment by dongjoon-hyun** [comment]
Thank you, @Yikun .

**Review by viirya — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon , @viirya , @Yikun .
Merged to master.

**Comment by dongjoon-hyun** [comment]
For the record, Java 21 images are published successfully.
- https://github.com/apache/spark-docker/actions/runs/10934045296
- https://hub.docker.com/layers/apache/spark/4.0.0-preview1-java21/images/sha256-5e0bb8becda30b59fcbd712dfac99aad22d50683c0df986a77f3db7c9d78e5e6?context=explore

**Pull request closed without merging**